code/game/objects/structures/mirror.dm

```diff
@@ -139,25 +139,25 @@
 	icon_state = "magic_mirror"
 	var/list/choosable_races = list()
 
-/obj/structure/mirror/magic/New()
+/obj/structure/mirror/magic/Initialize()
+	. = ..()
```

Comment:
Do something like this here:
```dm
/obj/structure/mirror/magic/Initialize()
	. = ..()
	setup_choosable_races()

/obj/structure/mirror/magic/proc/setup_choosable_races()
	if(!choosable_races.len)
		(...)
```

That way you can call parent first on Initialize and redefine that proc for `/obj/structure/mirror/magic/lesser` and the rest with their own logic that may come before.

Reply:
Very smart.